http/src/main/scala/org/broadinstitute/dsde/workbench/leonardo/monitor/DateAccessedUpdater.scala

```diff
@@ -45,21 +45,23 @@ class DateAccessedUpdater[F[_]](
       .compile
       .drain
 
-  private def updateDateAccessed(msg: UpdateDateAccessMessage): F[Unit] =
-    metrics.incrementCounter("jupyterAccessCount") >>
-      clusterQuery
-        .clearKernelFoundBusyDateByProjectAndName(msg.cloudContext, msg.runtimeName, msg.dateAccessd)
-        .flatMap(_ =>
-          clusterQuery.updateDateAccessedByProjectAndName(msg.cloudContext, msg.runtimeName, msg.dateAccessd)
-        )
-        .transaction
-        .void
+  private def updateDateAccessed(msg: UpdateDateAccessMessage): F[Unit] = msg.updateTarget match {
+    case UpdateTarget.Runtime(runtimeName) =>
+      metrics.incrementCounter("jupyterAccessCount") >>
+        clusterQuery
+          .clearKernelFoundBusyDateByProjectAndName(msg.cloudContext, runtimeName, msg.dateAccessd)
+          .flatMap(_ => clusterQuery.updateDateAccessedByProjectAndName(msg.cloudContext, runtimeName, msg.dateAccessd))
+          .transaction
+          .void
+    case UpdateTarget.App(appName) =>
+      appQuery.updateDateAccessed(appName, msg.dateAccessd).transaction.void
+  }
 }
 
 object DateAccessedUpdater {
   implicit val updateDateAccessMessageOrder: Ordering[UpdateDateAccessMessage] =
     Ordering.fromLessThan[UpdateDateAccessMessage] { (msg1, msg2) =>
-      if (msg1.cloudContext == msg2.cloudContext && msg1.runtimeName == msg2.runtimeName)
+      if (msg1.cloudContext == msg2.cloudContext && msg1.toString == msg2.toString)
```

Comment:
since UpdateTarget's members are case classes, they have `equals` and `hashCode` already defined, right? I think it'd be more stable to use `msg1.cloudContext == msg2.cloudContext && msg1.updateTarget == msg2.updateTarget` here if that's true

Reply:
good point :)